neon-mmd/websurfx
# 🛠️ Proper working implementation for the `hybrid` caching feature

## What does this PR do?

This PR fixes the `hybrid` caching implementation proposed in PR #223 by providing a proper working solution.

## Why is this change important?

This change is essential as it fixes the `hybrid` caching feature proposed in PR #223.

## How to test this PR locally?

It can be tested by installing and running `Websurfx` by following the steps provided below:

``` shell
git clone https://github.com/neon-mmd/websurfx.git
cd websurfx
cargo build -r --features redis-cache
```

After installing the `Websurfx` by following the steps mentioned above. Then launch the app and redis-server by following the given steps below:

``` shell
redis-server --port 8082 &
./target/release/websurfx
```

After that provide different search queries in the browser and see if everything works and then just shutdown the `redis-server` and then test the search queries again and see if everything works as expected.

## Author's checklist

- [x] Fix the `hybrid` caching implementation proposed in PR #223.
- [x] Provide several improvements to documentation to the docs. 
- [x] Bump the app version to `v0.21.1`.

## Related issues

Closes #244
